### DISCUSSION TITLE
new self hosted install hangs on `search`

### DISCUSSION THREAD

**Discussion by mingfang** [Budibase/budibase#5114] [Help]
On my newly install, I was able to create a new admin and was able to create a new app.
But when trying to load the app, my chrome shows calls to `search` are timing out.

`https://budibase-example.rebelsoft.com/api/ta_21d33191642740259a52f9733976529e/search`

I'm not seeing anything obvious in the worker or apps logs.
How can I best troubleshoot this?

**Comment by shogunpurple** [comment]
Hi @mingfang - is this running on docker-compose? 

It seems like there could be an issue reading from couchDB here. Could you tell us a bit more about what you are trying to query? Ie. you can see the data inside the builder data UI, but not in your app? Or can you not query the rows at all?

**Comment by mingfang** [comment]
hi @shogunpurple 
my install was roughly based on the official docker-compose install, but built using my own Terraform modules.
The modules are available here https://github.com/mingfang/terraform-k8s-modules/tree/master/examples/budibase

As for the hanging problem, all I did after install was to create a new project based on any of the "quick start templates", say "Agency Client Portal".  After creating the project, it opens to the editor and that's where it hangs/times out.
The query payload looks like this
```json
{
"query":{
  "string":{},
  "fuzzy":{},
  "range":{},
  "equal":{},
  "notEqual":{},
  "empty":{},
  "notEmpty":{},
  "contains":{},
  "notContains":{}
},
"bookmark":null,
"limit":10,
"sort":null,
"sortOrder":"ascending",
"sortType":"string",
"paginate":true
}
```

I'll leave the instance running and you can login to see the problem.
https://budibase-example.rebelsoft.com/
login as `user@example.com`/`password`

**Comment by shogunpurple** [comment]
Thanks a lot for the detail @mingfang - this is super helpful. Also really nice job on your project there - some really nice terraform modules for different products. I'll definitely use this for my own projects - so just gave you your 50th star :)

So, everything on the budibase side looks good from looking at your terraform modules. Your NGINX configuration is in order, and from logging into your app only calls to the `/search` endpoint are timing out. It's a 504 from NGINX, which means that NGINX is having a hard time getting a response from the upstream.

I think I know what the problem is. When running on kubernetes, we rely on CouchDB search capabilities, provided by this image:
https://hub.docker.com/r/kocolosk/couchdb-search 

The CouchDB helm chart provides options to turn this on, which you can see in our `values.yaml` here:
https://github.com/Budibase/budibase/blob/master/charts/budibase/values.yaml#L211

I then looked at your CouchDB terraform module here, and noticed that it seems to be missing any kind of search configuration:
https://github.com/mingfang/terraform-k8s-modules/tree/master/modules/couchdb

This is likely to be the problem - if you enable search support in your couchDB module, and then have the budibase tf module pass this config down to your couch module with search enabled, it _should_ work.

Thanks!

**Comment by mingfang** [comment]
@shogunpurple 
I searched the entire `budibase` repo for `enableSearch` and I only found it in `values.yaml`.
It appears that flag is ignored.
Is there a way to disable search altogether and avoid this problem?

**Reply by shogunpurple** [reply]
Hi @mingfang - the `enableSearch` flag gets passed down into the `couchdb` helm chart. There's no way to disable search altogether, as the platform relies on it to function correctly.

**Reply by egimbernat** [reply]
I'm having this same issue. I'm using the latest versions up to today. The logs of close give me:

`2022-07-27 18:11:02,744 scalang.node.ErlangHandler [INFO] channel disconnected org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext@22701e2f [id: 0x34d859ac, /127.0.0.1:40437 :> /127.0.0.1:41267] DISCONNECTED. peer: null 2022-07-27 18:11:02,745 scalang.node.ServerHandshakeHandler [ERROR] Channel closed during handshake 2022-07-27 18:11:02,745 scalang.node.ErlangHandler [DEBUG] handler message HandshakeFailed('couchdb@budibase-couchdb-0.budibase-couchdb.budibase-front.svc.cluster.local)`

**Reply by cptburke** [reply]
hi, i had the same problem. I fixed it by changing the couchdb container to ibmcom/couchdb3 in values.yaml for the helm install . See here https://github.com/Budibase/budibase/discussions/6879#discussioncomment-3221462

```yaml
couchdb:
  enableSearch: false
  image:
    repository: ibmcom/couchdb3
    tag: 3.2.1
```
